integration-tests/validation/src/main/java/com/android/designcompose/testapp/validation/examples/BrushTest.kt

```diff
@@ -77,40 +93,82 @@ class SizingShaderBrush(private val shader: RuntimeShader) : ShaderBrush() {
+    val brushState: MutableState<Brush> = remember { mutableStateOf(solidColor) }
+    CustomBrushTestDoc.Main(fill = brushState)
+
+    Row(modifier = Modifier.offset(y = 300.dp)) {
+        ElevatedTextButton(
+            onClick = {
+                shaderEnabled.value = false
+                brushState.value = solidColor
+            },
+            text = "Solid Blue"
+        )
+        ElevatedTextButton(
+            onClick = {
+                shaderEnabled.value = false
+                brushState.value = linearGradient
+            },
+            text = "Linear"
+        )
+        ElevatedTextButton(
+            onClick = {
+                shaderEnabled.value = false
+                brushState.value = radialGradient
+            },
+            text = "Radial"
         )
 
-    // Android T introduces AGSL and RuntimeShader. Robolectric (part of our test infrastructure)
-    // only supports software rendering.
-    val brush: () -> Brush =
+        // Android T introduces AGSL and RuntimeShader. Robolectric (part of our test
+        // infrastructure) only supports software rendering.
         if (
             Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU &&
                 "robolectric" != Build.FINGERPRINT
         ) {
+            val infiniteTransition = rememberInfiniteTransition(label = "animate shader")
+            val movingValue =
+                infiniteTransition.animateFloat(
+                    label = "moving value for shader",
+                    initialValue = 0.0f,
+                    targetValue = 10.0f,
+                    animationSpec =
+                        infiniteRepeatable(
+                            animation = tween(10 * 1000, easing = FastOutSlowInEasing),
+                            repeatMode = RepeatMode.Reverse
+                        )
+                )
             val shader = RuntimeShader(CUSTOM_SHADER)
-            val shaderBrush = SizingShaderBrush(shader)
-            // The kotlin compiler seems to get confused without semicolons on these statements.
-            shader.setFloatUniform("iTime", 0.0f);
-
-            {
-                // Only sample the state in the generator function; this means that we avoid
-                // recomposition and only do the redraw phase.
-                shader.setFloatUniform("iTime", movingValue.value)
-                shaderBrush
+            shader.setFloatUniform("iTime", 0.0f)
+            LaunchedEffect(shaderEnabled.value) {
+                if (shaderEnabled.value) {
+                    snapshotFlow { movingValue.value }
+                        .collect {
+                            shader.setFloatUniform("iTime", it)
+                            // Check if shader has been disabled. Race condition when click on
+                            // other buttons.
+                            if (shaderEnabled.value) {
+                                // Need to create new object to trigger redraw.
```

Comment:
Is this triggering just a redraw, or is it also triggering recomposition since the `MutableState` `brushState` is changing?

Reply:
It only triggers redraw because the state value is only read during redraw.

When I click the buttons, https://github.com/google/automotive-design-compose/blob/9ee3d5f24905ab093f6506106c6527e40da2837d/designcompose/src/main/java/com/android/designcompose/DesignFrame.kt#L699 and https://github.com/google/automotive-design-compose/blob/9ee3d5f24905ab093f6506106c6527e40da2837d/designcompose/src/main/java/com/android/designcompose/FrameRender.kt#L273 will re-run, but **NOT**  https://github.com/google/automotive-design-compose/blob/9ee3d5f24905ab093f6506106c6527e40da2837d/designcompose/src/main/java/com/android/designcompose/DesignFrame.kt#L62

For squoosh same that only the `squooshRender` and `squooshTextRender` reruns but not `squooshRoot`